File: Python/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ans = nums[0]
        t = [nums[0]]
        for i in range(1, len(nums)):
            t.append(max(t[-1] + nums[i], nums[i]))
            if t[-1] > ans:
                ans = t[-1]
        return ans

File: old/JavaScript/53-maximum-subarray.js
let maxSubArray = (nums) => {
    let evaluation = [];
    nums.forEach((num, key) => {
        if (key === 0) evaluation.push(num);
        else {
            evaluation.push(Math.max(evaluation[key - 1] + num, num));
        }
    });
    let max = evaluation[0];
    evaluation.forEach((item) => {
        if (item > max) max = item;
    });
    return max;
};
